src/CommunityToolkit.Aspire.Hosting.Azure.DataApiBuilder/DataApiBuilderHostingExtension.cs

```diff
@@ -13,25 +13,56 @@ public static class DataApiBuilderHostingExtension
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<DataApiBuilderContainerResource> AddDataAPIBuilder(this IDistributedApplicationBuilder builder,
+        [ResourceName] string name,
+        params string[] configFilePaths)
+    {
+        return builder.AddDataAPIBuilder(name, null, configFilePaths);
+    }
+
+    /// <summary>
+    /// Adds a DataAPIBuilder application to the application model. Executes the containerized DataAPIBuilder app.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
     /// <param name="port">The port number for the Data API Builder container.</param>"
+    /// <param name="configFilePaths">The path to the config or schema file(s) for Data API Builder.</param>"
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
     public static IResourceBuilder<DataApiBuilderContainerResource> AddDataAPIBuilder(this IDistributedApplicationBuilder builder,
         [ResourceName] string name,
-        string configFilePath = "./dab-config.json",
-        int? port = null)
+        int? port = null,
+        params string[] configFilePaths)
     {
         ArgumentNullException.ThrowIfNull("Service name must be specified.", nameof(name));
-        ArgumentNullException.ThrowIfNull("Config file path must be specified.", nameof(configFilePath));
 
         var resource = new DataApiBuilderContainerResource(name);
 
         var rb = builder.AddResource(resource)
-            .WithAnnotation(new ContainerImageAnnotation { Image = DataApiBuilderContainerImageTags.Image, Tag = DataApiBuilderContainerImageTags.Tag, Registry = DataApiBuilderContainerImageTags.Registry })
+            .WithAnnotation(new ContainerImageAnnotation
+            {
+                Image = DataApiBuilderContainerImageTags.Image,
+                Tag = DataApiBuilderContainerImageTags.Tag,
+                Registry = DataApiBuilderContainerImageTags.Registry
+            })
             .WithHttpEndpoint(port: port, targetPort: 5000, name: DataApiBuilderContainerResource.HttpEndpointName)
             .WithDataApiBuilderDefaults();
 
-        rb.WithBindMount(configFilePath, "/App/dab-config.json", true);
+        // Use default config file path if no paths are provided
+        if (configFilePaths is [])
+        {
+            configFilePaths = ["./dab-config.json"];
+        }
+
+        foreach (var configFilePath in configFilePaths)
+        {
+            var configFileName = File.Exists(configFilePath)
+                ? Path.GetFileName(configFilePath)
+                : throw new FileNotFoundException($"Config file not found: {configFilePath}");
+
+            rb.WithBindMount(configFilePath, $"/App/{configFileName}", true);
```

Comment:
Ah, I see that that won't work when it comes to the GraphQL file required with CosmosDB.

I think that we should handle Cosmos slightly differently though, since it's required that you provide `schema.graphql` when you do a Cosmos backend, we should enforce that in the API:

```csharp
public IResourceBuilder<DataApiBuilderContainerResource> WithReference(this IResourceBuilder<DataApiBuilderContainerResource> builder, CosmosResource resource, string schemaFile)
{
    return builder.WithResource(resource).WithBindMound(schemaFile, "/App/schema.graphql");
}
```

This way, from an API discoverability perspective, it's very obvious of what the requirements are.

Reply:
And, no, the name of the file is not accounted for anywhere in DAB.

Reply:
Isn't it? I thought that there was some logic to look for specific file names only, otherwise how does DAB know what to load?